src/NuGet.Clients/PackageManagement.VisualStudio/Migrate/ProjectJsonToPackageRefMigrator.cs

```diff
@@ -0,0 +1,146 @@
+                        _project.MSBuildProjectPath));
+                }
+                dependencies.AddRange(targetFramework.Dependencies);
+
+            }
+
+            dependencies.AddRange(packageSpec.Dependencies);
+            foreach (var dependency in dependencies)
+            {
+                var includeFlags = dependency.IncludeType;
+                var privateAssetsFlag = dependency.SuppressParent;
+                var metadataElements = new List<string>();
+                var metadataValues = new List<string>();
+                if (includeFlags != LibraryIncludeFlags.All)
+                {
+                    metadataElements.Add("IncludeAssets");
+                    metadataValues.Add(LibraryIncludeFlagUtils.GetFlagString(includeFlags));
+                }
+
+                if (privateAssetsFlag != LibraryIncludeFlagUtils.DefaultSuppressParent)
+                {
+                    metadataElements.Add("PrivateAssets");
+                    metadataValues.Add(LibraryIncludeFlagUtils.GetFlagString(privateAssetsFlag));
+                }
+                await _project.InstallPackageWithMetadataAsync(dependency.Name, dependency.LibraryRange.VersionRange, metadataElements, metadataValues);
+            }
+        }
+
+        private void MigrateRuntimes(PackageSpec packageSpec, Microsoft.Build.Evaluation.Project buildProject)
+        {
+            var runtimes = packageSpec.RuntimeGraph.Runtimes;
+            if (runtimes != null && runtimes.Count > 0)
+            {
+                var runtimeIdentifierValues = string.Join(";", runtimes.Keys);
+                buildProject.SetProperty("RuntimeIdentifiers", runtimeIdentifierValues);
+            }
+        }
+
+        private void RemoveProjectJsonReference(Microsoft.Build.Evaluation.Project buildProject)
+        {
+            var projectJsonItem = buildProject.GetItems("None")
+                .Where(t => string.Equals(t.EvaluatedInclude, Path.GetFileName(ProjectJsonFilePath), StringComparison.OrdinalIgnoreCase))
+                .FirstOrDefault();
+
+            if (projectJsonItem != null)
+            {
+                buildProject.RemoveItem(projectJsonItem);
+            }
+
+        }
+
+        private void CreateBackupAndSave(out string backupProjectFile, out string backupJsonFile)
+        {
+            var backupDirectory = Path.Combine(Path.GetDirectoryName(_project.MSBuildProjectPath), "Backup");
+            if(!Directory.Exists(backupDirectory))
+            {
+                Directory.CreateDirectory(backupDirectory);
+            }
+
+            backupJsonFile = Path.Combine(backupDirectory, Path.GetFileName(ProjectJsonFilePath));
+            File.Move(ProjectJsonFilePath, backupJsonFile);
+            backupProjectFile = Path.Combine(backupDirectory, Path.GetFileName(_project.MSBuildProjectPath));
+            File.Copy(_project.MSBuildProjectPath, backupProjectFile);
+            _envdteProject.Save();
```

Comment:
This has to be done on the UI thread. This should either use a JTF here or throw if not on the ui thread.